hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockDataStreamOutput.java

```diff
@@ -92,6 +92,19 @@
 
   private int chunkIndex;
   private final AtomicLong chunkOffset = new AtomicLong();
+
+  // Similar to 'BufferPool' but this list maintains only references
+  // to the ByteBuffers.
+  private List<ByteBuffer> bufferPool;
+
+  // List containing buffers for which the putBlock call will
```

Comment:
it seems weird and confusing why we need tow different lists referencing the same set of byteBuffers? Can you add more comments/explanation?